pkg/controller/node/controller_utils.go

```diff
@@ -77,9 +86,31 @@ func deletePods(kubeClient clientset.Interface, recorder record.EventRecorder, n
 		}
 		remaining = true
 	}
+
+	if len(updateErrList) > 0 {
+		return false, utilerrors.NewAggregate(updateErrList)
+	}
 	return remaining, nil
 }
 
+// setPodTerminationReason attempts to set a reason and message in the pod status, updates it in the apiserver,
+// and returns an error if it encounters one.
+func setPodTerminationReason(kubeClient clientset.Interface, pod *api.Pod, nodeName string) (*api.Pod, error) {
+	if pod.Status.Reason == node.NodeUnreachablePodReason {
+		return pod, nil
+	}
+
+	pod.Status.Reason = node.NodeUnreachablePodReason
+	pod.Status.Message = fmt.Sprintf(node.NodeUnreachablePodMessage, nodeName, pod.Name)
+
+	var updatedPod *api.Pod
+	var err error
+	if updatedPod, err = kubeClient.Core().Pods(pod.Namespace).UpdateStatus(pod); err != nil {
```

Comment:
Delete can only fail if you opt-in to a conditional delete.

Is it actually likely that the termination reason for an already terminated pod is likely to change?  Seems like a retryonconflict that makes sure the status reason and message haven't changed ought to do, right?

I could also see an argument for making this error non-fatal and just `utilruntime.HandleError` and move on.  Seems like that might be better than stopping this entirely.